packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -388,6 +388,33 @@ export function ordinalToArray(ord: string) {
     return a;
 }
 
+/**
+ * Checks if the sequence number is unassigned. Unassigned changes
+ * are always local changes that have not been round tripped to the
+ * server and assigned a real sequence number yet.
+ */
+function isUnassigned(seq: number | undefined): boolean {
+    return seq === UnassignedSequenceNumber;
+}
+
+/**
+ * This function takes a number of sequence numbers and normalizes them.
+ * Normalization makes the sequence numbers globally comparable to things
+ * like other seqs, ref seqs, and the min seq.
+ *
+ * Specifically this function assigns unassigned sequence numbers a temporary sequence number near
+ * Number.MAX_SAFE_INTEGER. This works as we know all unassigned sequence numbers will eventually
+ * be assigned a sequence number that is greater than any we can currently see. The temporary
+ * sequence numbers are assigned based on order, the highest seq will be given to the first, and so on.
+ * For instance in the case of a removed segment, it will have two sequences the insert sequence, and
+ * the removed sequence. A segment must be inserted before it can be removed, so the removed sequence
+ * should be the first element, and the insert the next, so the temporary removed sequence is always
+ * greater than the temporary insert sequence.
+ */
+function normalizeSequenceNumberIfUnassigned(...seqs: (number | undefined)[]): number[] {
+    return seqs.map((v,i)=>isUnassigned(v) ? Number.MAX_SAFE_INTEGER - i : v ?? 0);
+}
```

Comment:
BTW - It took me a while to catch on that you pass the inputs in reverse order [remove, insert] rather than [insert, remove], which makes the ' - i' work.

Reply:
It might be a bit more readable if "normalize" took arguments in increasing order of seq# (e.g., [insertSeq, removeSeq]) rather than the other way around.  (It took me a while to catch on that this is why '- i' worked.)

The last time I measured, destructuring arrays was slower than destructuring objects with named fields.  I don't think it would be measurable, but will mention this alternative anyway in case it's of interest:

const { seq0: insertSeq, seq1: removeSeq } = normalize(/* seq0: */ insertSeq, /* seq1: */ removeSeq);